hpbandster/utils.py

```diff
@@ -11,11 +11,30 @@
 
 
 
-def nic_name_to_host(nic_name):
-	""" translates the name of a network card into a valid host name"""
-	from netifaces import ifaddresses, AF_INET
-	host = ifaddresses(nic_name).setdefault(AF_INET, [{'addr': 'No IP addr'}] )[0]['addr']
-	return(host)
+def nic_name_to_host(nic_name=None):
+    """ translates the name of a network card into a valid host name"""
+    from netifaces import ifaddresses, AF_INET
+
+    def get_nic_name_from_system():
+        import re
+        import subprocess
+        process = subprocess.Popen("ip route get 8.8.8.8".split(),
+                                   stdout=subprocess.PIPE)
+        output = process.stdout.read().decode()
+        s = re.search(r'dev\s*(\S+)', output)
+        return s.group(1)
+
+    # if the network card name is not a valid one an ecxeption will be raised
+    # and the method get_nic_name_from_system will discover a valid card name
+    try:
```

Comment:
I like the sentiment of your functionality, but I assumed that most users would call the nic_name_to_host function and catch the exception themselves in case they need to try more nic names because they are different across the cluster (which is usually not the case).